HW04.Rmd

```diff
@@ -30,34 +34,113 @@ Knitr is part of the knitr package. Allows you to change column names and add ca
 
 ```{r month, warning=FALSE}
 
+month_abbrev <-as.factor(month.abb) #creates a factor class vector of the built-in character vector containing abbrevations of the months
+
+gun_deaths_months_kable <- gun_deaths %>% 
+    select(year, month) %>% 
+    group_by(month) %>% 
+    count(month)
+
+gun_deaths_months_kable$month <- month_abbrev #changes the numbers that represent the months into their abbreviations
+
+kable(gun_deaths_months_kable, format = "markdown", col.names = c("Month", "# of Deaths"), caption = "Sum of deaths for each month from 2012-2014")
+
 ```
 
 
+
+
 ### Generating a bar chart with human-readable labels on the x-axis. That is, each month should be labeled "Jan", "Feb", "Mar" (full or abbreviated month names are fine), not `1`, `2`, `3`.
 
 ```{r month_chart}
-
+gun_deaths_months_kable$month = factor(gun_deaths_months_kable$month, levels = month.abb) # to set the order of the factor
+gun_deaths_months_kable %>% 
+    ggplot(aes(x = month, y = n)) +
+    geom_col() +
+    labs(title = "Sum of deaths for each month from 2012-2014",
+         x = "Month",
+         y = "Number of Deaths")
 ```
 
 ## Generating a bar chart that identifies the number of gun deaths associated with each type of intent cause of death. The bars should be sorted from highest to lowest values.
 
 ```{r intent}
-
+gun_deaths_intent <- gun_deaths %>% 
+    select(intent) %>% 
+    count(intent) %>% 
+    arrange(desc(n))
+
+gun_deaths_intent$intent = factor(gun_deaths_intent$intent)
+
+gun_deaths_intent %>% 
+    ggplot(aes(x = reorder(intent, -n), y = n)) + #using the reorder() function with aes() allows for the reordering of the x-axis based on y-value
+    geom_col() +
+    labs(title = "Gun Deaths by Intent",
+         x = "Intent",
+         y = "# of Deaths")
 ```
 
 ## Generating a boxplot visualizing the age of gun death victims, by sex. Print the average age of female gun death victims.
 
 ```{r age_&_sex}
+gun_death_age_sex <- gun_deaths %>% 
+    select(age, sex) %>% 
+    group_by(sex, age) %>% 
+    filter(age != "Not Available") %>% 
+    type.convert() #This break in the code forces the age column to be numeric
+    
+gun_death_age_sex2 <- gun_death_age_sex %>% 
+    count(age) %>% 
+    arrange(age)
+
+gun_death_age_sex2 %>% 
+    ggplot(aes(x = sex, y = age)) +
+    geom_boxplot() +
+    labs(title = "Age at Death of Gun Victims by Sex, 2012-2014",
+         x = "Sex",
+         y = "Age")
+    
+```
+
+This code shows that the average age of death for female victims of gun violence is ~43.7 years old. 
+```{r}
+gun_death_age_sex3 <- gun_death_age_sex %>% 
+    filter(sex == "F") %>% 
+    ungroup(sex, age) %>% 
+    summarize(average_age_F = mean(age))
+
+print(gun_death_age_sex3$average_age_F[1])
+```
+
+
 
+I kind of feel like I didn't do this last graph correctly, so I decided to graph this data in another way as well. 

```

Comment:
Ooo I really like this graph; it gives more information than the box plot because it has the added component of number of deaths, but both graphs are effective in communicating that the age distribution is similar for both sexes.